### ISSUE TITLE
[BUG] Crash On Random Name Generation (ArrayIndexOutOfBounds: 26)

### ISSUE THREAD

**Issue by UnknownEvil** [evolvio/evolv.io#135]
The bug appears to be in the getRandomChar() function. (see below)
```
java.lang.ArrayIndexOutOfBoundsException: 26
	at evolv.io.NameGenerator.getRandomChar(NameGenerator.java:89)
	at evolv.io.NameGenerator.newName(NameGenerator.java:17)
	at evolv.io.Creature.createName(Creature.java:72)
	at evolv.io.Creature.<init>(Creature.java:63)
	at evolv.io.Creature.<init>(Creature.java:47)
	at evolv.io.Board.maintainCreatureMinimum(Board.java:553)
	at evolv.io.Board.iterate(Board.java:365)
	at evolv.io.EvolvioColor.draw(EvolvioColor.java:68)
	at processing.core.PApplet.handleDraw(PApplet.java:2418)
	at processing.awt.PSurfaceAWT$12.callDraw(PSurfaceAWT.java:1540)
	at processing.core.PSurfaceNone$AnimationThread.run(PSurfaceNone.java:316)
```

Originally I'd though it was fixed, but it's not.
I've just run it over night, and it crashed.

(It looks like the random name generation is done poorly either way, although I haven't spent enough time to actually check exactly how it works)

**Comment by NateUniAccount** [comment]
Just done some debugging, it happens when the random letterFactor is very high eg. 99.99999. It looks like the cause is inaccurate floating point arithmetic as the letterFactor ends up being a very tiny value when it should be zero. I'll put a fix in.

**Comment by NateUniAccount** [comment]
Fixed with bugfix branch #136

**Comment by UnknownEvil** [comment]
Alright, I'll close this issue then.
